### PR TITLE
Add a way to get patterns from the PropertyCondition

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Adds a way to get the property condition patterns, since I see many people making patterns similar to PropertyCondition but can't use the PropertyCondition register method due to the pattern.

This allows developers to collect and use the patterns in the way they want.
```java
String[] patterns = Streams.concat(
		Arrays.stream(getPatterns(PropertyType.CAN, property, type)),
		Arrays.stream(getPatterns(PropertyType.IS, property, type)))
	.toArray(String[]::new);
Skript.registerCondition(condition, ConditionType.PROPERTY, patterns);
```

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java [hunk 1]
```diff
@@ -64,6 +64,12 @@ public enum PropertyType {
 		 */
 		BE,
 
+		/**
+		 * Indicates that the condition is in a form of <code>something is/are something</code>,
+		 * also possibly in the negated form
+		 */
+		IS,
+
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: this is just BE, why does it need to be added

**Review by sovdeeth — CHANGES_REQUESTED** [review]
looks good otherwise

File: src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java (hunk 1 above)
Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: BE doesn't equal is/are

Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: well is/are are conjugated forms of "to be", so BE makes sense

Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: Is this blocking to not have both enums?

Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: Yeah, either rename BE or keep BE (i prefer keeping)
It's just confusing to have two types that do and mean the exact same thing

Comment (Pikachu920) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: renaming would be a breaking change!

Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: Not renaming, it's adding the proper terminology.

File: src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java [hunk 2]
```diff
@@ -86,49 +92,66 @@ public enum PropertyType {
...
+	/**
+	 * Returns the patterns for the given property type, property and type.
+	 * 
+	 * @param propertyType the property type, see {@link PropertyType}
+	 * @param property the property name, for example <i>fly</i> in <i>players can fly</i>
+	 * @param type must be plural, for example <i>players</i> in <i>players can fly</i>
+	 * @return patterns formmated for {@link Skript#registerCondition(Class, String...)}
+	 */
+	public static String[] getPatterns(PropertyType propertyType, String property, String type) {
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:126]: i'd prefer this be private. if someone wants to get the patterns they could use the syntaxelementinfo

Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:126]: I'd like it to be public so other syntaxes can use the property patterns without being property conditions, and without duplicating code.

Comment (Pikachu920) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:126]: image

File: src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java (hunk 1 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:72]: As already mentioned, BE is the proper terminology. No need to add another that does the same thing.

**Review by sovdeeth — CHANGES_REQUESTED** [review]

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java [hunk 3]
```diff
@@ -86,49 +86,65 @@ public enum PropertyType {
...
 		if (type.contains("%"))
 			throw new SkriptAPIException("The type argument must not contain any '%'s");
 
 		switch (propertyType) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java:124]: This can be replaced with returning a switch expression

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**